homeassistant/components/squeezebox/media_player.py

```diff
@@ -126,18 +140,75 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     lms = Server(async_get_clientsession(hass), host, port, username, password)
     known_servers.add(ipaddr)
 
-    players = await lms.async_get_players()
-    if players is None:
-        raise PlatformNotReady
-    media_players = []
-    for player in players:
-        media_players.append(SqueezeBoxDevice(player))
+    async def _discovery(now=None):
```

Comment:
Does this need to be nested in `async_setup_entry`?

Reply:
Yes. It is much easier for the player discovery functions to be nested - than we can use `async_call_later` without parameters. This thread is specific to the configuration entry (the individual Logitech Media Server).